src/gsNurbs/gsBSpline.hpp

```diff
@@ -127,6 +128,62 @@ gsBSpline<T> gsBSpline<T>::segmentFromTo(T u0, T u1, T tolerance) const
   return gsBSpline<T>(Basis(give(knotsRes)), give(coefRes));
 }
 
+template<class T>
+std::vector<gsBSpline<T>> gsBSpline<T>::toBezier(T tolerance) const {
+  std::vector<gsBSpline<T>> bezierSegments;
+  const size_t numSegments = this->knots(0).size() - 1; // Assuming this is valid
+  bezierSegments.reserve(numSegments);
+
+  gsBSpline<T> currentSegment(*this);
+  gsBSpline<T> leftPart;
+
+  for (auto iter = this->knots().ubegin() + 1; iter != this->knots().uend() - 1; ++iter) {
+    currentSegment.splitAt(*iter, leftPart, currentSegment, tolerance);
+    bezierSegments.push_back(std::move(leftPart));
+  }
+
+  bezierSegments.push_back(std::move(currentSegment)); // Add the last segment
+  return bezierSegments;
+}
+
+template<class T>
+std::vector<internal::gsCurveIntersectionResult<T>> gsBSpline<T>::intersect(const gsBSpline<T>& other,
+                                                    T tolerance) const {
+  std::vector<internal::gsBoundingBoxPair<T>> hulls = internal::getPotentialIntersectionRanges<T>(*this, other);
+
+  std::vector<internal::gsCurveIntersectionResult<T>> results;
+  for (const auto &hull : hulls) {
+    gsBSpline<T> crv1 = this->segmentFromTo(hull.b1.getRange().getMin(), hull.b1.getRange().getMax());
+    gsBSpline<T> crv2 = other.segmentFromTo(hull.b2.getRange().getMin(), hull.b2.getRange().getMax());
+
+    internal::gsCurveCurveDistanceSystem<T> obj(crv1, crv2);
+    gsMatrix<T, 2, 1> uv;
+    uv(0, 0) = 0.5 * (crv1.domainStart() + crv1.domainEnd());
+    uv(1, 0) = 0.5 * (crv2.domainStart() + crv2.domainEnd());
+    T distance = obj.compute(uv, tolerance);
+
+    if (distance < std::max(1e-10, tolerance)) {
```

Comment:
please use math::max instead of std::max
and math::min instead of std::min
everywhere

Reply:
Done!